features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -142,25 +142,26 @@ internal class Base64Serializer private constructor(
         drawableHeight: Int,
         displayMetrics: DisplayMetrics,
         imageWireframe: MobileSegment.Wireframe.ImageWireframe,
-        base64SerializerCallback: Base64SerializerCallback?
-    ): Bitmap? {
+        base64SerializerCallback: Base64SerializerCallback
+    ) {
         drawableUtils.createBitmapOfApproxSizeFromDrawable(
             drawable,
             drawableWidth,
             drawableHeight,
-            displayMetrics
-        )?.let { resizedBitmap ->
-            serializeBitmapAsynchronously(
-                drawable,
-                bitmap = resizedBitmap,
-                shouldCacheBitmap = true,
-                imageWireframe,
-                base64SerializerCallback
-            )
-            return resizedBitmap
-        }
-
-        return null
+            displayMetrics,
+            base64SerializerCallback = base64SerializerCallback,
+            bitmapCreationCallback = object : BitmapCreationCallback {
```

Comment:
I see that `Base64Serializer` already has an executor. I wonder why don't you use this instead to executor call/execute the `drawableUtils.createBitmapOfApproxSizeFromDrawable` and leave that method synchronous and remove the `LambdaExecutor`